File: extension.ts
//% weight=70 icon="\u237E" color=#74DF00 block="MyBlock"
namespace myblock {

    //% blockId=myblock_getn
    //% block="getn"
    export function getn(): number {
    	basic.showString("sim:getn")
        return 123
    }

    //% blockId=myblock_setstr
    //% block="setstr|%str"
    export function setstr(str: string): void {
    	basic.showString("sim:setstr")
    	basic.showString(str)
    }
}

File: extension.cpp
#include "pxt.h"

namespace myblock {

//% blockId=myblock_getn
//% block="get |%n"
//% shim=myblock::getn
int getn() {
	return 432;
}

//% blockId=myblock_initjson
//% block="Json Init|string %json"
//% shim=myblock::initjson
void fcnnfromjson(String json) {

	const char *jsonStr = PXT_STRING_DATA(json);
    uBit.serial.send(jsonStr);
}


}
